### ISSUE TITLE
Can't create a view for Notification instance. Please register a view provider.

### ISSUE THREAD

**Issue by k2b6s9j** [steelbrain/package-deps#1]
Stack Trace:
```
Error: Can't create a view for Notification instance. Please register a view provider.
    at ViewRegistry.module.exports.ViewRegistry.createView (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/src/view-registry.js:96:15)
    at ViewRegistry.module.exports.ViewRegistry.getView (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/src/view-registry.js:65:21)
    at new PackageDepsView (/Users/kepler/.atom/packages/linter-ruby/node_modules/atom-package-deps/lib/view.js:18:38)
    at Function.installPackage (/Users/kepler/.atom/packages/linter-ruby/node_modules/atom-package-deps/lib/main.js:32:18)
    at Function.install (/Users/kepler/.atom/packages/linter-ruby/node_modules/atom-package-deps/lib/main.js:27:26)
    at Object.activate (/Users/kepler/.atom/packages/linter-ruby/lib/main.js:21:34)
    at Package.module.exports.Package.activateNow (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/src/package.js:243:19)
    at /opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/src/package.js:224:30
    at Package.module.exports.Package.measure (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/src/package.js:167:15)
    at Package.module.exports.Package.activate (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/src/package.js:216:14)
    at PackageManager.module.exports.PackageManager.activatePackage (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/src/package-manager.js:486:21)
    at /opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/src/package-manager.js:469:29
    at Config.module.exports.Config.transact (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/src/config.js:324:16)
    at PackageManager.module.exports.PackageManager.activatePackages (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/src/package-manager.js:464:19)
    at PackageManager.module.exports.PackageManager.activate (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/src/package-manager.js:445:46)
    at Atom.module.exports.Atom.startEditorWindow (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/src/atom.js:670:21)
    at Object.<anonymous> (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/src/window-bootstrap.js:12:8)
    at Object.<anonymous> (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/src/window-bootstrap.js:23:4)
    at Module._compile (module.js:452:26)
    at Object.defineProperty.value [as .js] (/opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/src/compile-cache.js:169:21)
    at Module.load (module.js:347:32)
    at Function.Module._load (module.js:302:12)
    at Module.require (module.js:357:17)
    at require (module.js:376:17)
    at setupWindow (file:///opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/static/index.js:82:25)
    at window.onload (file:///opt/homebrew-cask/Caskroom/atom/latest/Atom.app/Contents/Resources/app.asar/static/index.js:35:9)
```

Upon first-run of an instance with only AtomLinter/linter-ruby@d172f17 installed.

![screen shot 2015-09-05 at 6 53 41 pm](https://cloud.githubusercontent.com/assets/594124/9701884/f434597a-53ff-11e5-9907-54273c5ad337.png)

**Comment by steelbrain** [comment]
That is super-super weird but I'll investigate.

**Comment by k2b6s9j** [comment]
Good thing this was `linter-ruby`. :wink:

**Comment by steelbrain** [comment]
@k2b6s9j I am unable to repro. Can you make sure that you are on the latest version of atom?
This snippet executes without an error for me, does it produce an error for you?

```
a = atom.notifications.addError("some Error")
console.log(atom.views.getView(a))
```

**Comment by k2b6s9j** [comment]
That snippet returns an `atom-notification` element in the console...

**Comment by steelbrain** [comment]
@k2b6s9j the code that's producing the error above, is the same code as this one. Probably that was a temporary weird error. Would you mind trying the linter-ruby thing again?

**Comment by k2b6s9j** [comment]
Reloaded the window. Still happening.

**Comment by steelbrain** [comment]
@k2b6s9j does this work? https://github.com/steelbrain/package-deps/pull/2

**Comment by k2b6s9j** [comment]
How do I point NPM to that revision?

**Comment by steelbrain** [comment]
Well, you clone the repo in node_modules folder and checkout that branch

**Comment by k2b6s9j** [comment]
Ah. Thanks.

**Comment by k2b6s9j** [comment]
ReferenceError: packageName is not defined
    at /Users/kepler/.atom/packages/linter-ruby/node_modules/atom-package-deps/lib/view.js:17:70

**Comment by steelbrain** [comment]
@k2b6s9j just pushed. Pull it and try again.

**Comment by k2b6s9j** [comment]
Error no longer occurs. It now shows an Info notification, with a progress bar. But the progress bar doesn't seem to be making any progress. However, APM installs have always been kinda slow for me, I'll give it a minute or two.

**Comment by k2b6s9j** [comment]
![screen shot 2015-09-05 at 8 47 53 pm](https://cloud.githubusercontent.com/assets/594124/9702170/69abce40-540f-11e5-96bb-e125435762b3.png)

**Comment by k2b6s9j** [comment]
Okay. It's making no indication of any progress.

**Comment by steelbrain** [comment]
@k2b6s9j the progress bar doesn't make any progress for one package, it's like 0% to 100% at once because apm doesn't report it's install progress. However if you have more than one dependencies, then it'll mark one package installed as a progress point. I think I'll make the progress bar animate a bit.

**Comment by k2b6s9j** [comment]
Okay.

**Comment by k2b6s9j** [comment]
I think the install has stalled @steelbrain it still hasn't moved.

**Comment by steelbrain** [comment]
@k2b6s9j can you check if `linter-ruby` exists in `~/.atom/packages`?
There's no reason for it to be stalled, it worked fine for me on the first run :stuck_out_tongue: 
But yes, we do need Unit Tests.

**Comment by k2b6s9j** [comment]
`linter-ruby` exists in `~/.atom/packages`, `linter` does not. I'll restart Atom though and see if that works.

**Comment by k2b6s9j** [comment]
Still not installing.

**Comment by steelbrain** [comment]
@k2b6s9j writing unit tests. Give me a few minutes.

**Comment by k2b6s9j** [comment]
Awesome. Alright.

**Comment by steelbrain** [comment]
A confirmed fix is in v2.0.1
